jenkins/Jenkinsfile.j2

```diff
@@ -407,503 +99,29 @@ def make(docker_type, path, make_flag) {
 {% set microtvm_template_projects = ['build/microtvm_template_projects',] %}
 s3_prefix = "tvm-jenkins-artifacts-prod/tvm/${env.BRANCH_NAME}/${env.BUILD_NUMBER}"
 
+// General note: Jenkins has limits on the size of a method (or top level code)
+// that are pretty strict, so most usage of groovy methods in these templates
+// are purely to satisfy the JVM
+{% include "jenkins/Prepare.groovy.j2" %}
```

Comment:
is there a script you run to regenerate this file? i always invoke from a venv inside of jenkins (though I guess maybe that's not necessary now that we've added it to the docker containers). in any case, now that we have paths involved, i think it would be good to fully capture the build recipe here.

Reply:
this is a template file, not a generated one (`.j2` == Jinja2), it get used in the same way as the main jenkinsfile:

```bash
python jenkins/generate.py
```

all the paths in the templates are relative to the repo root though due to how that script calls Jinja2:

https://github.com/apache/tvm/blob/85e42b6af38ea3bd0c99c8208d7baed5086a8959/jenkins/generate.py#L89